proto/dataplane.proto

```diff
@@ -0,0 +1,210 @@
+syntax = "proto3";
+
+package config;
+
+option go_package = "dataplane/grpc";
+
+service ConfigService {
+  rpc GetConfig (GetConfigRequest) returns (GatewayConfig);
+  rpc GetConfigGeneration (GetConfigGenerationRequest) returns (GetConfigGenerationResponse);
+  rpc UpdateConfig (UpdateConfigRequest) returns (UpdateConfigResponse);
+}
+
+message GetConfigRequest {
+}
+
+message UpdateConfigRequest {
+  GatewayConfig config = 1;
+}
+
+message UpdateConfigResponse {
+  Error error = 1;
+  string message = 2;
+}
+
+enum Error {
+  ERROR_NONE = 0;
+  ERROR_VALIDATION_FAILED = 1;
+  ERROR_APPLY_FAILED = 2;
+  ERROR_UNKNOWN_ERROR = 3;
+}
+
+message GetConfigGenerationRequest {
+}
+
+message GetConfigGenerationResponse {
+  uint64 generation = 1;
+}
+
+/* ================================================ */
+/* ==== Interfaces (both underlay and overlay) ==== */
+/* ================================================ */
+
+message Interface {
+  string name = 1;
+  string ipaddr = 2;
+  IfType type = 3;
+  IfRole role = 4;
+}
+
+enum IfType {
+  IF_TYPE_ETHERNET = 0;
+  IF_TYPE_VLAN = 1;
+  IF_TYPE_LOOPBACK = 2;
+  IF_TYPE_VTEP = 3;
+}
+
+enum IfRole {
+  IF_ROLE_FABRIC = 0;
+  IF_ROLE_EXTERNAL = 1;
+}
+
+/* ================= */
+/* ==== Overlay ==== */
+/* ================= */
+
+message PeeringIPs {
+  oneof rule {
+    string cidr = 1;
+    string not = 2;
+  }
+}
+
+message PeeringAs {
+  oneof rule {
+    string cidr = 1;
+    string not = 2;
+  }
+}
+
+message Expose {
+  repeated PeeringIPs ips = 1;
+  repeated PeeringAs as = 2;
+}
+
+message PeeringEntryFor {
+  string vpc = 1;
+  repeated Expose expose = 2;
+}
+
+message VpcPeering {
+  string name = 1; /* unique: e.g. VPC-1--VPC-2 */
+  repeated PeeringEntryFor for = 2;
```

Comment:
Can we add a comment saying that this repeated must actually be 2 PeeringEntryFor's ?

Reply:
In fact, I would not even use repeated, but "one" and "two" or "left", "right"

Reply:
done
let's have it repeated and make only a pair in dataplane